mtp_noms_ops/apps/settings/views.py

```diff
@@ -152,3 +152,38 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['prisons'] = self.request.user_prisons
         return context
+
+
+class JobInformationView(FormView):
+    title = _('Help us improve this service')
+    template_name = 'settings/job-information.html'
+    success_url = reverse_lazy('security:dashboard')
+    form_class = JobInformationForm
+
+    def get_success_url(self):
+        if 'next' in self.request.GET:
+            return '{path}?{query}'.format(
+                path=self.success_url,
+                query=urlencode({'next': self.request.GET['next']})
+            )
+        return self.success_url
+
+    def form_valid(self, *args, **kwargs):
+        if has_provided_job_information(self.request.user):
+            return redirect(self.get_success_url())
+
+        if self.form_class(self.request.POST).is_valid():
```

Comment:
this is a little weird. the method's signature is actually `form_valid(self, form)` so you already have a _valid_ form variable to work with